contractsV5/kyberDAO/KyberDAO.sol

```diff
@@ -0,0 +1,649 @@
+        startBlock = camp.startBlock;
+        endBlock = camp.endBlock;
+        totalKNCSupply = camp.totalKNCSupply;
+        formulaParams = camp.formulaParams;
+        link = camp.link;
+        options = camp.options;
+    }
+
+    function getCampaignVoteCountData(uint campID) public view returns(uint[] memory voteCounts, uint totalVoteCount) {
+        uint[] memory votes = campOptionPoints[campID];
+        if (votes.length == 0) {
+            return (voteCounts, totalVoteCount);
+        }
+        totalVoteCount = votes[0];
+        voteCounts = new uint[](votes.length - 1);
+        for (uint i = 0; i < voteCounts.length; i++) {
+            voteCounts[i] = votes[i + 1];
+        }
+    }
+
+    function getCampaignOptionVoteCount(uint campID, uint optionID)
+        public view
+        returns(uint voteCount, uint totalVoteCount)
+    {
+        uint[] memory voteCounts = campOptionPoints[campID];
+        if (voteCounts.length == 0 || optionID == 0 || optionID >= voteCounts.length) {
+            return (voteCount, totalVoteCount);
+        }
+        voteCount = voteCounts[optionID];
+        totalVoteCount = voteCounts[0];
+    }
+
+    function getCampaignWinningOptionAndValue(uint campID)
+        public view
+        returns(uint optionID, uint value)
+    {
+        if (!isCampExisted[campID]) { return (0, 0); } // not existed
+
+        Campaign storage camp = campaignData[campID];
+
+        // not found or not ended yet, return 0 as winning option
+        if (camp.endBlock == 0 || camp.endBlock > block.number) { return (0, 0); }
+
+        bool hasConcluded;
+        (hasConcluded, optionID) = decodeWinningOptionData(winningOptionData[campID]);
+        if (hasConcluded) {
+            if (optionID == 0 || optionID >= camp.options.length) {
+                // no winning option or invalid winning option
+                return (0, 0);
+            }
+            return (optionID, camp.options[optionID]);
+        }
+
+        uint totalSupply = camp.totalKNCSupply;
+        // no one has voted in this epoch
+        if (totalSupply == 0) { return (0, 0); }
+
+        uint[] memory voteCounts = campOptionPoints[campID];
+
+        // Finding option with most votes
+        uint winningOption = 0;
+        uint maxVotedCount = 0;
+        for (uint i = 1; i < voteCounts.length; i++) {
+            if (voteCounts[i] > maxVotedCount) {
+                winningOption = i;
+                maxVotedCount = voteCounts[i];
+            } else if (voteCounts[i] == maxVotedCount) {
+                winningOption = 0;
+            }
+        }
+        // more than 1 options have same vote count
+        if (winningOption == 0) { return (0, 0); }
+
+        FormulaData memory formulaData = decodeFormulaParams(camp.formulaParams);
+
+        uint totalVotes = voteCounts[0];
+        // compute voted percentage (in precision)
+        uint votedPercentage = totalVotes.mul(PRECISION).div(camp.totalKNCSupply);
+
```

Comment:
Nice to add comment:
`votedPercentage below minimum treshold`

Reply:
ok I added few more comments here